src/helpers.js

```diff
@@ -62,6 +67,70 @@ function readFileString (file) {
   })
 }
 
+/**
+ * Load local private key
+ * @private
+ * @param certName name of the private key
+ * @returns {Buffer| undefined} privateKey
+ */
+function loadPrivateKey (certName) {
+  if (certName) {
+    const certLocation = path.join(CONFIG_PATH, certName)
+    if (fs.existsSync(certLocation)) {
+      aioLogger.debug('Private key loaded from: %s', certLocation)
+      return fs.readFileSync(path.join(CONFIG_PATH, certName))
+    }
+  }
+}
+/**
+ * Fetch new private key.
+ * @private
+ * @param {string} env IMS environment type
+ * @param {string} certName Certificate name that can be used to verify the signature
+ * @returns {Promise<string>} privateKey
+ */
+async function getPrivateKey (env, certName) {
+  if (env && certName) {
+    const URL = `https://static.adobelogin.com/keys/${env}/${certName}`
+    const certLocation = path.join(CONFIG_PATH, certName)
+    aioLogger.debug('Fetching a new private key...')
+    const fetchRetry = new HttpExponentialBackoff()
+    const res = await fetchRetry.exponentialBackoff(URL, {})
+    const data = await res.text()
+    /* istanbul ignore else */
+    if (isPrivateKey(data)) {
+      fs.writeFileSync(certLocation, data)
+      aioLogger.debug('Private key saved at: %s', certLocation)
+      return data
+    }
+  }
+  aioLogger.debug('Invalid params %s, %s', env, certName)
+  return Promise.reject(new Error('Invalid params.'))
+}
+
+/**
+ * Validates provided token using an IMS public key cert.
+ *
+ * @param {string} token Jwt token you want to validate. It validates only the access_token type.
+ * @returns {Promise<void>} Will throw if token is invalid
+ */
+async function verifyJwt (token) {
```

Comment:
The whole implementation is incorrect. This should not use jwt.verify (call it offlineTokenValidate or something to that effect), but verify the token according to the 6 clauses in the "Validating a token" section of the internal wiki page.